src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -114,11 +113,17 @@ def __init__(self, api_func: Callable, **kwargs: Any) -> None:
     def _fetch(self) -> str:
         """ Execute the requested api method as a one-off fetch"""
         response = self.api_func(**self.kwargs)
-        # metadata is a client.V1ListMeta object type
-        metadata = response.metadata  # type: client.V1ListMeta
-        self._items = {item.metadata.name: item for item in response.items}
-        log.info('Full fetch of {}. result: {}'.format(self.api_func, len(self._items)))
-        return metadata.resource_version
+        # try-except pattern because customObjectApi objects aren't subscriptable
+        try:
+            metadata = response['metadata']
+            self._items = {item['metadata']['name']: item for item in response['items']}
+            log.info('Full fetch of {}. result: {}'.format(self.api_func, len(self._items)))
+            return metadata['resourceVersion']
+        except TypeError:
+            metadata = response.metadata
+            self._items = {item.metadata.name: item for item in response.items}
+            log.info('Full fetch of {}. result: {}'.format(self.api_func, len(self._items)))
+            return metadata.resource_version
```

Comment:
Why do we need to do this? .. ( i think we can encapsutate the retrieval of information from resources like "local.storage.openshift.io")
and in the worst case.... we can check what type of object is metadata  and return the dict element, or the attribute without repeating the same lines twice.

Reply:
The reason we need this is because the Kubernetes Python CustomObjectsApi returns a response that isn't "subscriptable" which, from what I understand, means you can't address it's element using `map[key]` but you can use `map.key`, I agree that this might not be the best solution however.

Reply:
So it seems that we have another good reason to encapsulate the call and results of "list_cluster_custom_object" into a class that provide a public method to return a dict of discovered devices with the attributes we use in the orchestrator.
But that implies to make more generic the class KubernetesResources, and move all the devices stuff into a "bridge" class , and create specific classes to deal with  storage classes created by different operators.
So probably more easy and simple your suggestion! :-)